tools/tm-signer-harness/internal/test_harness_test.go

```diff
@@ -0,0 +1,207 @@
+package internal
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/privval"
+	"github.com/tendermint/tendermint/types"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/tendermint/crypto/ed25519"
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+const (
+	keyFileContents = `{
+	"address": "D08FCA3BA74CF17CBFC15E64F9505302BB0E2748",
+	"pub_key": {
+		"type": "tendermint/PubKeyEd25519",
+		"value": "ZCsuTjaczEyon70nmKxwvwu+jqrbq5OH3yQjcK0SFxc="
+	},
+	"priv_key": {
+		"type": "tendermint/PrivKeyEd25519",
+		"value": "8O39AkQsoe1sBQwud/Kdul8lg8K9SFsql9aZvwXQSt1kKy5ONpzMTKifvSeYrHC/C76Oqturk4ffJCNwrRIXFw=="
+	}
+}`
+
+	stateFileContents = `{
+	"height": "0",
+	"round": "0",
+	"step": 0
+}`
+
+	genesisFileContents = `{
+	"genesis_time": "2019-01-15T11:56:34.8963Z",
+	"chain_id": "test-chain-0XwP5E",
+	"consensus_params": {
+		"block_size": {
+		"max_bytes": "22020096",
+		"max_gas": "-1"
+		},
+		"evidence": {
+		"max_age": "100000"
+		},
+		"validator": {
+		"pub_key_types": [
+			"ed25519"
+		]
+		}
+	},
+	"validators": [
+		{
+		"address": "D08FCA3BA74CF17CBFC15E64F9505302BB0E2748",
+		"pub_key": {
+			"type": "tendermint/PubKeyEd25519",
+			"value": "ZCsuTjaczEyon70nmKxwvwu+jqrbq5OH3yQjcK0SFxc="
+		},
+		"power": "10",
+		"name": ""
+		}
+	],
+	"app_hash": ""
+}`
+
+	defaultAcceptDeadline = 100
+	defaultConnDeadline   = 100
+)
+
+func TestRemoteSignerTestHarnessMaxAcceptRetriesReached(t *testing.T) {
+	cfg := makeConfig(t, 1, 2)
+	defer cleanup(cfg)
+
+	th, err := NewTestHarness(log.TestingLogger(), cfg)
+	assert.NoError(t, err)
+	donec := make(chan struct{})
```

Comment:
why run in a goroutine if you block right after?

Reply:
Fixed 👍